src/components/fav-button/fav-button.jsx

```diff
@@ -0,0 +1,71 @@
+import PropTypes from "prop-types";
+import { Button } from "react-bootstrap";
+import { useState, useEffect } from "react";
+import { API } from "../../utils/links";
+
+export const FavButton = ({ movieID }) => {
+  const userData = JSON.parse(localStorage.getItem("userData"));
+  const token = localStorage.getItem("token");
+  const [isFavorited, setFavorited] = useState(false);
+
+  useEffect(() => {
+    if (userData.favorites.includes(movieID)) {
+      setFavorited(true);
+    } else {
+      setFavorited(false);
+    }
```

Comment:
⛏️You could refactor this  code like:
```suggestion
    const isFavoriteMovie = userData.favorites.includes(movieID);
    setFavorited(isFavoriteMovie);
```

Reply:
Thank you so much for your extensive feedback! I have implemented your suggestions.